src/main/java/com/solr/clientwrapper/domain/service/SolrCoreService.java

```diff
@@ -0,0 +1,226 @@
+    public SolrResponseDTO createCore(String coreName) {
+        //HttpSolrClient solrClient = new HttpSolrClient.Builder("http://localhost:8983/solr").build();
+        CoreAdminRequest.Create request = new CoreAdminRequest.Create();
+        request.setCoreName(coreName);
+        request.setInstanceDir(coreName);
+        request.setConfigSet("_default");
+        request.setDataDir("data");
+
+        SolrResponseDTO solrResponseDTO=new SolrResponseDTO();
+        solrResponseDTO.setCoreName(coreName);
+
+        try {
+            CoreAdminResponse coreAdminResponse =request.process(solrClient);
+
+            solrResponseDTO.setStatusCode(200);
+            solrResponseDTO.setMessage("Successfully created Solr Core: "+coreName);
+
+        } catch (SolrServerException e) {
+            e.printStackTrace();
+
+            solrResponseDTO.setStatusCode(400);
+            solrResponseDTO.setMessage("Unable to create Solr Core: "+coreName+". SolrServerException.");
+
+        } catch (IOException e) {
+            e.printStackTrace();
+
+            solrResponseDTO.setStatusCode(400);
+            solrResponseDTO.setMessage("Unable to create Solr Core: "+coreName+". IOException.");
+
+        }catch (Exception e){
+            e.printStackTrace();
+
+            solrResponseDTO.setStatusCode(400);
+            solrResponseDTO.setMessage("Unable to create Solr Core: "+coreName+". Exception.");
+
+        }
+
+        return solrResponseDTO;
+    }
+
+    @Override
+    public SolrResponseDTO renameCore(String coreName, String newName) {
+
+        SolrResponseDTO solrResponseDTO=new SolrResponseDTO();
+        solrResponseDTO.setCoreName(coreName);
+
+        try {
+            CoreAdminResponse coreAdminResponse=CoreAdminRequest.renameCore(coreName,newName,solrClient);
+
+            solrResponseDTO.setStatusCode(200);
+            solrResponseDTO.setMessage("Successfully renamed Solr Core: "+coreName+" to "+newName);
+
+        } catch (SolrServerException e) {
+            e.printStackTrace();
+
+            solrResponseDTO.setStatusCode(400);
+            solrResponseDTO.setMessage("Unable to rename Solr Core: "+coreName+" to "+newName+". SolrServerException.");
+
+        } catch (IOException e) {
+            e.printStackTrace();
+
+            solrResponseDTO.setStatusCode(400);
+            solrResponseDTO.setMessage("Unable to rename Solr Core: "+coreName+" to "+newName+". IOException.");
+
+        }catch (Exception e){
+            e.printStackTrace();
+
+            solrResponseDTO.setStatusCode(400);
+            solrResponseDTO.setMessage("Unable to rename Solr Core: "+coreName+" to "+newName+". Exception.");
+
+        }
+
+        return solrResponseDTO;
+    }
+
+    @Override
+    public SolrResponseDTO deleteCore(String coreName, boolean deleteIndex, boolean deleteDataDir, boolean deleteInstanceDir) {
+        CoreAdminRequest.Unload request=new CoreAdminRequest.Unload(true);
+        request.setCoreName(coreName);
```

Comment:
these 3 lines are used in all methods 
make seperate method  to set request object and use it everywhere

Reply:
These 3 lines are different in different methods. Please do let me know if you mean any other changes.